### PR TITLE
refactor(not-found): introduce a proper and unique 'notFound' value

### PR DESCRIPTION
**Pull request by serras**
(Either this of #7 should be rebased in top of the other before merging)

After many discussions about whether to use `null` or `undefined`, I finally saw the light: [`Symbol`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) 🌟 

MDN explains that:

> A symbol value may be used [...] as an implementation-supported unique identifier in general

Exactly what we need to differentiate non-existing values in optionals from any other value!

### CODE REVIEW

**Comment by serras** [comment]
The branch has been rebased, and conflicts have been fixed

File: __tests__/functions.test.js [hunk 1]
```diff
@@ -9,15 +9,15 @@ describe('Function Operators', () => {
     expect(exp(5)(3)).toBe(125)
   })
 
-  test('isNil -> should check if a value is null or undefined', () => {
+  /* test('isNil -> should check if a value is null or undefined', () => {
```
Comment (kutyel) [inline __tests__/functions.test.js]: Should we keep the `isNil` function or not? 🤔

**Review by kutyel — CHANGES_REQUESTED** [review]
Please rebase again and fix conflicts 🙏

File: __tests__/functions.test.js (hunk 1 above)
Comment (serras) [reply __tests__/functions.test.js]: I don't see the benefit, so I've removed it.

**Comment by serras** [comment]
Rebased and test passed (locally)!

File: __tests__/functions.test.js (hunk 1 above)
Comment (kutyel) [inline __tests__/functions.test.js]: Remove the tests please instead of commenting it and run prettier 😋

File: src/Optional.js [hunk 2]
```diff
@@ -36,7 +37,13 @@ class Optional {
...
-export const optionalIx = (index) => optional(get(index), set(index))
+export const optionalProp = (key) => optional(
+  (obj)      => key in obj ? obj[key] : notFound,
+  (val, obj) => key in obj ? { ...obj, [key]: val } : obj
+)
+
+// optionalIx : Number → Optional s a
+export const optionalIx = (index) => optional(
+  (obj)      => index in obj ? obj[index] : notFound,
+  (val, obj) => index in obj ? { ...obj, [index]: val } : obj
```
Comment (kutyel) [inline src/Optional.js]: This is a very haskelly formatting... we have prettier for this 😝  please run it!

**Pull request merged**